### PR TITLE
This prevents a crash due to instance-config.php not existing

### PR DESCRIPTION
**Pull request by ReCaffeinated**
The installer will now check if the file exists and if not creates the file

This is important, there's a small enough developer base for lainchan as is, we don't need to make contributing code any harder than it already is, I spent like eight hours just setting lainchan up
(granted I've never web developed before)
and I want to make sure nobody has the same problem from here on out

### CODE REVIEW

**Comment by barrucadu** [comment]
I may be missing something, but I'm not sure why this patch is necessary. The installer [already checks if the file is present/writable](https://github.com/lainchan/lainchan/blob/master/install.php#L733), and [gives you a warning](https://github.com/lainchan/lainchan/blob/master/templates/installer/check-requirements.html#L22) if not.

What exactly is the error you got?

**Comment by barrucadu** [comment]
Oh, I see. [`loadConfig` fails if the file isn't there](https://github.com/lainchan/lainchan/blob/master/inc/functions.php#L120), and is [called unconditionally](https://github.com/lainchan/lainchan/blob/master/inc/functions.php#L31) in functions.php, which is included [by the installer](https://github.com/lainchan/lainchan/blob/master/install.php#L6). Not fun.

File: install.php [hunk 1]
```diff
@@ -3,6 +3,11 @@
 // Installation/upgrade file	
 define('VERSION', '4.9.93');
 
+if (fopen('inc/instance-config.php' , 'a') === false) {
+	print('install.php does not have permission to write to /inc/, without permission the installer cannot continue');
+	exit();
+	}
```
Comment (barrucadu) [inline install.php]: Outdent this `}`, otherwise looks good.

**Comment by ReCaffeinated** [comment]
Yeah it was tricky, I wanted to contribute code to the site but due to this error I had trouble getting it installed, and needed a lot of help from the mumble server.  So I guess this'll be my first submitted patch.  I'm inexperienced at large projects but I've been fiddling with code for a long time now, and I learn quick

**Comment by barrucadu** [comment]
Well, in vichan (and I assume tinyboard), the file always *does* exist, it just gets overwritten when you run the installer. When we put lainchan on github, we made the decision to ignore instance-config.php, as we obviously had our configuration in there. But then we haven't exactly tested running the installer since then.

**Pull request merged**